db/db_impl_secondary.cc

```diff
@@ -184,6 +249,37 @@ Status DBImplSecondary::RecoverLogFiles(
         continue;
       }
       WriteBatchInternal::SetContents(&batch, record);
+      std::vector<uint32_t> column_family_ids;
+      status = CollectColumnFamilyIdsFromWriteBatch(batch, &column_family_ids);
+      if (status.ok()) {
+        SequenceNumber seq = versions_->LastSequence();
+        for (const auto id : column_family_ids) {
+          ColumnFamilyData* cfd =
+              versions_->GetColumnFamilySet()->GetColumnFamily(id);
+          if (cfd == nullptr) {
+            continue;
+          }
+          if (cfds_changed->count(cfd) == 0) {
+            cfds_changed->insert(cfd);
+          }
+          auto curr_log_num = std::numeric_limits<uint64_t>::max();
+          if (current_log_.count(cfd) > 0) {
+            curr_log_num = current_log_[cfd];
+          }
+          if (!cfd->mem()->IsEmpty() &&
```

Comment:
Might be good to add a comment here to briefly explain why we need a new memtable

Reply:
We need a new memtable because the previous one is marked immutable and added to the immutable list. It's similar to and much simpler than `SwitchMemtable`.